### PR TITLE
Update vendor/golang.org/x/sys

### PR DESCRIPTION
**Pull request by tuxillo**
Please ignore this PR for now (see #4817). I am trying to use the CI to see if updating vendor/golang.org/x/sys builds fine as pointed out by @techknowlogick .

### CODE REVIEW

**Comment by techknowlogick** [comment]
Thanks for PR ❤️ You may also need to commit the lock file.

Can you try running `make vendor` too. that may help clear up the CI Fail. I'm sorry for not giving you the correct instructions.

**Comment by tuxillo** [comment]
@techknowlogick I don't know what I'm doing wrong. These are my steps:

- Modify Gopkg.toml
- Run 'make vendor' and commit all the files it produces on vendor/
- Run 'make test-vendor' and no errors are emitted.

The CI is generating a patch, is that supposed to be generated by some script and committed prior pushing?

**Comment by sapk** [comment]
@tuxillo 
edit Gopkg.toml to  follow master
[[constraint]]
  branch = "master"
  name = "golang.org/x/sys"

Execute :
```
dep ensure -update golang.org/x/sys
git add Gopkg.toml Gopkg.lock vendor/golang.org/x/sys
```

and you should be good to commit and push to PR.

From what you have already done, I think you just forgot to add change of Gopkg.lock (git add Gopkg.lock)

**Comment by tuxillo** [comment]
@sapk no luck. I tried exactly with your instructions with both branch and revision but none worked. I must be missing something.

The results of the failure show as if some files should not be there.

**Comment by sapk** [comment]
@tuxillo I tested those commands in #5169 but the diff of your lock file seems strange. For exemple digest where removed and not updated. What version of dep are you using ?

**Comment by tuxillo** [comment]
Let me describe my process.

I'm using Ubuntu 18.04.1 LTS. So from https://docs.gitea.io/en-us/install-from-source :

- go get -d -u code.gitea.io/gitea
- cd ~/go/src/code.gitea.io/gitea/
- git checkout master
- git pull
- git checkout -b patch-1

And then from there I do the instructions you guys provided. dep version is: 

ii  go-dep                                                      0.3.2-2                                     amd64        Go dependency management tool

    $ dep version
    dep:
     version     : devel
     build date  : 
     git hash    : 
     go version  : go1.8.3
     go compiler : gc
     platform    : linux/amd64

**Comment by sapk** [comment]
Try to do `go get -u github.com/golang/dep/cmd/dep` to update to the latest version of dep.

**Comment by tuxillo** [comment]
@sapk that did it! Thanks.

**Review by sapk — APPROVED** [review]

File: Gopkg.toml
Comment (sapk) [inline Gopkg.toml]: Can you use the constraint "master" ? The commit 4497e2df6f9e69048a54498c7affbbec3294ad47 is 3 week old. This is not blocking for the PR but why not updating to the latest.
```suggestion
   branch = "master"
```

**Comment by sapk** [comment]
The constraint is only used for `dep ensure -update` or `dep status` ortherwise `dep ensure` will only follow the lock file. For example you can update to master the toml and let the lock to the current commit it shouldn't complain.

**Comment by techknowlogick** [comment]
@tuxillo Thanks to @sapk explaining to me in another PR, what he has suggested is the better course of action. I was too cautious when there was no need for it.

**Comment by sapk** [comment]
For information, the constraint where lock to a specific commit at the migration from the previous vendor tool to limit the number of change during the migration.

**Review by techknowlogick — APPROVED** [review]

**Comment by tuxillo** [comment]
@sapk @techknowlogick I've just done the change including the master branch. Please check.

**Comment by techknowlogick** [comment]
😄 Thanks for the PR

Looks like #5169 will probably be merged before this one, but I'll be sure to merge this one right after

**Comment by tuxillo** [comment]
Excellent! Thanks a lot for guiding me.
I think after updating x/sys gitea will build in some systems where it didn't before. NetBSD and SmartOS at the very least.

**Comment by techknowlogick** [comment]
@tuxillo alright. This has been merged. Thank you again for the PR 😄 Hopefully it works for the platforms that you need it to.

**Pull request merged**
